### PR TITLE
Add context to fatal OpenErrors

### PR DESCRIPTION
**Pull request by RossSmyth**
Hello, I'm new to Zig and had some frustrations earlier today building Zig and passing `--zig-lib-dir` then having "FileNotFound" error returned, but not being sure what file (or directory it seems) wasn't being found. This pull replaces the `try` on `openDir()` methods with a `catch` that if `FileNotFound`, will call `fatal()` with the directory that was supposed to be found formatted in.

I checked the call hierarchy and all of these errors should have lead to an exit of the compiler regardless, except without `fatal()` they would've just bubbled up to `main()` without any context. One thing I am unsure of is whether the paths formatted in should be absolute or relative. I did absolute when convenient, but otherwise just formatted in what was attempted to be opened.

One thing not done is add any testing of these exits. Based upon the [testMissingOutputPath()](https://github.com/ziglang/zig/blob/86ebd4b975d2f1e01f468e06daab6e28c06f9719/test/cli.zig#L144) test it seems that is desired. Also these catches should be done for `openFile()` methods as well to provide more context.

Thanks for your work, and any help/critique you can provide.

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -3092,7 +3174,14 @@ pub fn cmdFmt(gpa: *Allocator, args: []const []const u8) !void {
     defer fmt.out_buffer.deinit();
 
     for (input_files.items) |file_path| {
-        try fmtPath(&fmt, file_path, check_flag, fs.cwd(), file_path);
+        fmtPath(&fmt, file_path, check_flag, fs.cwd(), file_path) catch |err| {
```
Comment (Vexu) [inline src/main.zig:3177]: This doesn't seem correct, `fmtPath` already handles everything except `AccessDenied` which should be handled in `fmtPathDir` and not be fatal.

**Review by Vexu — CHANGES_REQUESTED** [review]
I don't see any reason to limit this to just `FileNotFound`, unless doing so broke something else.

File: src/main.zig (hunk 1 above)
Comment (RossSmyth) [reply src/main.zig:3177]: Good point, I missed that in my first read.

**Comment by RossSmyth** [comment]
> 
> 
> I don't see any reason to limit this to just `FileNotFound`, unless doing so broke something else.

After analyzing the possible errors that can returned I agree. I'll change that and the above change when I get home.

**Review by Vexu — APPROVED** [review]
One thing that could still be improved is that a lot of these directories have specific names e.g. local cache directory, standard library directory, which could also be added to the error messages.

**Review by Vexu — COMMENTED** [review]
Error messages are supposed to be lower case but there does seem to be other places in the compiler where this isn't the case yet.

**Comment by RossSmyth** [comment]
Should be fixed now

**Comment by Vexu** [comment]
Thanks!

**Pull request merged**